src/kafka/protocol/messages.js

```diff
@@ -301,19 +284,20 @@ const decodePayload = (input, { asParsed = true } = {}) => {
  * @returns {(Protocol~Message\|Protocol~Message[])} - messages with decoded payload
  */
 
-exports.decodeMessages = (messages) => {
+const decodeMessages = (messages) => {
   const decodeMessage = (message) => {
-    let payload = decodePayload(message.value.content.payload)
-    message.value.content.payload = payload
-    return message
+    let decodedMessage = clone(message)
+    let payload = decodePayload(decodedMessage.value.content.payload)
+    decodedMessage.value.content.payload = payload
+    return decodedMessage
   }
 
   if (Array.isArray(messages)) {
     let result = []
     for (let message of messages) {
       console.log(message.value.content)
```

Comment:
Please update console.log to use the Logger :D. Sorry only picked this one up now.